src/dataflow/src/render/mod.rs

```diff
@@ -154,6 +156,9 @@ pub struct RenderState {
     pub dataflow_tokens: HashMap<GlobalId, Box<dyn Any>>,
     /// Sender to give data to be cached.
     pub caching_tx: Option<mpsc::UnboundedSender<CacheMessage>>,
+    /// Frontier of sink writes (all subsequent writes will be at times at or
+    /// equal to this frontier)
+    pub sink_write_frontiers: HashMap<GlobalId, Rc<RefCell<Antichain<Timestamp>>>>,
 }
```

Comment:
I'm actively working against the stashing of externalized state here. Is there any way to not foist the complexity of this on to the renderer?